presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/GeometryUtils.java

```diff
@@ -73,6 +74,42 @@ public static GeometryTypeName valueOf(String type)
         }
     }
 
+    public static Envelope deserializeEnvelope(Slice shape)
+    {
+        if (shape == null) {
+            return null;
+        }
+        BasicSliceInput input = shape.getInput();
+
+        input.readInt(); // spatial reference ID
+
+        Envelope overallEnvelope = null;
+        while (input.available() > 0) {
+            int length = input.readInt();
+            ByteBuffer buffer = input.readSlice(length).toByteBuffer().slice().order(LITTLE_ENDIAN);
+            int type = buffer.getInt();
+            Envelope envelope;
+            if (type == 1) {    // point
+                double x = buffer.getDouble();
+                double y = buffer.getDouble();
+                envelope = new Envelope(x, y, x, y);
+            } else {
+                double xmin = buffer.getDouble();
+                double ymin = buffer.getDouble();
+                double xmax = buffer.getDouble();
+                double ymax = buffer.getDouble();
+                envelope = new Envelope(xmin, ymin, xmax, ymax);
+            }
+            if (overallEnvelope == null) {
+                overallEnvelope = envelope;
+            } else {
+                overallEnvelope.merge(envelope);
+            }
+        }
+
```

Comment:
ensure `overallEnvelope != null`?

Reply:
overallEnvelope can be null if geometry is empty (e.g. 'POINT EMPTY', 'POLYGON EMPTY', etc.); I'll add @Nullable to the signature